src/coins.cpp

```diff
@@ -1025,6 +1025,35 @@ bool CCoinsViewCache::UpdateScInfo(const CScCertificate& cert, CBlockUndo& block
     return true;
 }
 
+bool CCoinsViewCache::IsOutputMature(const uint256& txHash, unsigned int pos) const
+{
+    if (HaveCoins(txHash) == 0)
+        return false;
+
+    const CCoins* pRefCoin =  AccessCoins(txHash);
+
+    if (!pRefCoin->IsAvailable(pos))
```

Comment:
Consider testing ptr before dereferencing it, something like: if (!pRefCoin || !pRefCoin->IsAvailable(pos))